lib/Driver/ToolChains.cpp

```diff
@@ -1191,6 +1193,14 @@ ToolChain::constructInvocation(const GeneratePCHJobAction &job,
   addInputsOfType(Arguments, context.InputActions, file_types::TY_ObjCHeader);
   context.Args.AddLastArg(Arguments, options::OPT_index_store_path);
 
+  if (context.Args.hasFlag(options::OPT_static_executable,
+                   options::OPT_no_static_executable, false) ||
+      context.Args.hasFlag(options::OPT_static_stdlib, options::OPT_no_static_stdlib,
+                   false) ||
+      getTriple().isOSBinFormatWasm()) {
+    Arguments.push_back("-use-static-resource-dir");
+  }
```

Comment:
[memo] These lines should be upstreamed because it causes invalid module-cache hash in clang-importer when using -static-stdlib option even though other platforms.